packages/compass-schema-validation/src/modules/zero-state.js

```diff
@@ -77,6 +79,7 @@ export const changeZeroState = (isZeroState) => {
     const namespace = state.namespace;
 
     if (isZeroState === false) {
+      track('Schema Validation Edited', {});
```

Comment:
We might want to check with @mmarcon that this is tracking the right thing? On the one hand, this matches the legacy rule that it’s supposed to match, and it does make sense to track this. On the other hand, I’m surprised because, based on the event name, I would have thought this refers to actually editing the validation rules and the comment about debouncing would have matched that impression.

Reply:
Is this event collected like this only about when someone adds validation rules when they don't have any? Do I get this right? It's very possible I made a mistake when I went through the old events and map them into the new ones. Can we track separately when:
 * validation is added when there isn't any
 * existing validation rules are edited

As a side note, is it possible to track when validation rules that are added are written with JSON Schema (`$jsonSchema`) vs when they are written in regular MQL?

Reply:
> Is this event collected like this only about when someone adds validation rules when they don't have any? Do I get this right? 

@mmarcon Yes, that’s exactly.

@mabaasit Maybe let’s name this `Schema Validation Added` then?

> Can we track separately when:
> 
>     * validation is added when there isn't any
> 
>     * existing validation rules are edited

I think that’s possible. The tracking plan doc talks about about debouncing, but really, I think we’d only want to track this once while the validation tab is open, and then not track it again until the user has left the tab and come back to it?

> As a side note, is it possible to track when validation rules that are added are written with JSON Schema (`$jsonSchema`) vs when they are written in regular MQL?

I think we could track whether the validation rules contain `$jsonSchema`, yes.

Reply:
Ah – went through the events again, and understood why I wrote the part about debouncing.

I think `rulesadded` was tracked every time someone types anything inside the validation box.
`saved` was tracked when someone actually clicks the save button.

Yes, I think what you say about only tracking it once makes sense.